include/BrainRegions.hpp

```diff
@@ -0,0 +1,580 @@
+template <>
+class TDP43_inclusions<2> : public SeedingRegion<2>
+{
+public:
+  TDP43_inclusions()
+  {}
+
+  virtual bool
+  check_region(const Point<2> &p) const override
+  {
+    return ((p[0] > -15.0 && p[0] < -2.0) && (p[1] > 25.0 && p[1] < 50.0));
+  }
+};
+
+/**
+ * Factory function to create seeding regions.
+ * The seeding region is specified by the user in the main.cpp file.
+ *
+ * @param region Name of the seeding region.
+ * @return Unique pointer to the seeding region.
+ * @throw std::invalid_argument if the seeding region name is invalid.
+ */
+template <int dim>
+std::unique_ptr<SeedingRegion<dim>>
+getSeedingRegion(std::string region)
+{
+  if (region == "Tau inclusions")
+    return std::make_unique<Tau_inclusions<dim>>();
+  else if (region == "Amyloid-Beta deposits")
+    return std::make_unique<Amyloid_Beta_deposits<dim>>();
+  else if (region == "TDP-43 inclusions")
+    return std::make_unique<TDP43_inclusions<dim>>();
+  else
+    throw std::invalid_argument("Invalid seeding region");
+}
+
+/**
+ * Class to define the grey matter region in the brain.
+ *
+ * @tparam dim Dimension of the problem.
+ */
+template <int dim>
+class Grey_matter
+{
+public:
+  Grey_matter()
+  {}
+
+  /**
+   * Check if a point is inside the grey matter region.
+   *
+   * @param p Point to check.
+   * @return True if the point is inside the grey matter region, false otherwise.
+   */
+  bool
+  check_grey_matter(const Point<dim> &p)
+  {
+    return false;
+  }
+};
+
+/**
+ * Specialization of the Grey_matter class for the 3D domain
+ * "brain-h3.0.msh".
+ */
+template <>
+class Grey_matter<3>
+{
+public:
+  Grey_matter()
+  {}
+
+  bool
+  check_grey_matter(const Point<3> &p)
+  {
+    return ((p[0] < 33 || p[0] > 70) || (p[1] < 25 || p[1] > 120) ||
+            (p[2] > 85));
+  }
+};
+
+/**
+ * Specialization of the Grey_matter class for the 2D domain "ellipse.msh".
+ */
+template <>
+class Grey_matter<2>
+{
+public:
+  Grey_matter()
+  {}
+
+  bool
+  check_grey_matter(const Point<2> &p)
+  {
+    return (p[0] * p[0]) / (a * a) + (p[1] * p[1]) / (b * b) > 1.0;
+  }
+
+private:
+  // semi-axes length of the ellipse representing the grey matter region in 2D.
+  static constexpr double a = 60.0;
+  static constexpr double b = 40.0;
+};
+
+
+/**
+ * Class to define the Radial direction of the fibers in the brain, with respect
+ * to the global center of the domain.
+ *
+ * @tparam dim Dimension of the problem.
+ */
+template <int dim>
+class RadialDirection : public Function<dim>
+{
+public:
+  RadialDirection(const Point<dim> &global_center_)
+    : global_center(global_center_)
+  {}
+
+  virtual void
+  vector_value(const Point<dim> &p, Vector<double> &values) const override
+  {
+    Vector<double> radial(dim);
+    for (unsigned int i = 0; i < dim; ++i)
+      radial[i] = p[i] - global_center[i];
+    radial /= radial.l2_norm();
+
+    values[0] = radial[0];
+    values[1] = radial[1];
+    if (dim == 3)
+      values[2] = radial[2];

```

Comment:
The following would be dimension independent in a more flexible way:
```suggestion
    for (unsigned int i = 0; i < dim; ++i)
      values[i] = radial[i];
```